src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -22,50 +24,44 @@
 METADATA_TABLE = "defillama_stablecoins_metadata"
 
 
-def get_data(session, url):
-    start = time.time()
-    resp = session.request(
-        method="GET",
-        url=url,
-        headers={"Content-Type": "application/json"},
-    ).json()
-    log.info(f"Fetched from {url}: {time.time() - start:.2f} seconds")
-    return resp
-
-
-def process_breakdown_stables(data):
-    peg_type = data["pegType"]
-    balances = data["chainBalances"]
-
-    rows = []
-    for chain, balance in balances.items():
-        tokens = balance["tokens"]
-
-        for datapoint in tokens:
-            row = {}
-            row["chain"] = chain
-            row["dt"] = dt_fromepoch(datapoint["date"])
-
-            try:
-                row["circulating"] = datapoint["circulating"][peg_type]
-            except KeyError:
-                row["circulating"] = None
+def process_breakdown_stables(
+    data: dict, days: int = 30
+) -> Tuple[pl.DataFrame, Dict[str, Optional[str]]]:
+    """
+    Processes breakdown data for stablecoins, filtering for the most recent N days.
 
-            try:
-                row["bridged_to"] = datapoint["bridgedTo"][peg_type]
-            except KeyError:
-                row["bridged_to"] = None
+    Args:
+        data: The breakdown data from the API.
+        days: The number of days to filter in the processed data.
 
-            try:
-                row["minted"] = datapoint["minted"][peg_type]
-            except KeyError:
-                row["minted"] = None
+    Returns:
+        A tuple containing a Polars DataFrame with breakdown data and a metadata dictionary.
+    """
+    peg_type: str = data.get("pegType", "")
+    balances: Dict[str, dict] = data.get("chainBalances", {})
```

Comment:
I think we shouldn't have defaults, we want to make sure the data is exactly as we expect it to be otherwise we might end up ingesting some invalid data.